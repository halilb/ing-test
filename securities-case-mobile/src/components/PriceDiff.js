import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const PriceDiff = ({ diff }) => {
  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: diff < 0 ? '#F1374A' : '#55EDB5',
        },
      ]}
    >
      <Text style={styles.diff}>{diff}%</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: 52,
    alignItems: 'center',
    borderRadius: 2,
    paddingVertical: 2,
  },
  diff: {
    color: 'white',
    fontSize: 12,
  },
});

PriceDiff.propTypes = {
  diff: PropTypes.number.isRequired,
};

export default PriceDiff;
