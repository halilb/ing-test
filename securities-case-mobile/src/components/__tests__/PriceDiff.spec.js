import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import PriceDiff from '../PriceDiff';

it('renders a negative diff correctly', () => {
  const snapshot = renderer.create(<PriceDiff diff={-0.44} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

it('renders a positive diff correctly', () => {
  const snapshot = renderer.create(<PriceDiff diff={2.32} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

it('renders zero diff correctly', () => {
  const snapshot = renderer.create(<PriceDiff diff={0} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
