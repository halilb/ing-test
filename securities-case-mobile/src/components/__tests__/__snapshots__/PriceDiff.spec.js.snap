// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a negative diff correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 2,
        "paddingVertical": 2,
        "width": 52,
      },
      Object {
        "backgroundColor": "#F1374A",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
      }
    }
  >
    -0.44
    %
  </Text>
</View>
`;

exports[`renders a positive diff correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 2,
        "paddingVertical": 2,
        "width": 52,
      },
      Object {
        "backgroundColor": "#55EDB5",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
      }
    }
  >
    2.32
    %
  </Text>
</View>
`;

exports[`renders zero diff correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 2,
        "paddingVertical": 2,
        "width": 52,
      },
      Object {
        "backgroundColor": "#55EDB5",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
      }
    }
  >
    0
    %
  </Text>
</View>
`;
